### ISSUE TITLE
adapter+storage: storage collections initialized w/ read frontiers in advance of read holds

### ISSUE THREAD

**Issue by sploiselle** [MaterializeInc/materialize#14136]
### What version of Materialize are you using?

 v0.26.1-dev (4604a289e)

### How did you install Materialize?

Built from source

### What is the issue?

The following command:

```sql
materialize=> create table m (a int);
CREATE TABLE
```

generates the following log:

```
environmentd: 2022-08-09T01:45:55.515701Z ERROR mz_adapter::coord::read_policy: Storage collection User(1) has read frontier Antichain { elements: [1660009555457] } not less-equal to read_hold.time: 1660009555229
```

The source which generates this error says:
> This error *SHOULD* be fatal, but it is not currently maintained and cannot merge as an assert.

This seems like it is either is an acceptable situation and the comment/warning level should be amended, or this should be investigated.

**Comment by uce** [comment]
@jkosh44 should have a fix up for this in https://github.com/MaterializeInc/materialize/pull/14092.

**Comment by philip-stoev** [comment]
Also seen with `CREATE INDEX` and the test case for gh 13879 , if one requires a reproduction scenario.

**Comment by jkosh44** [comment]
Quick update, I closed #14092 and @frankmcsherry is planning on pushing a fix for this.

**Comment by frankmcsherry** [comment]
I have something that might reduce the noise, but there would still be a "bug" in that we don't know the full space of reasons this triggers. In particular, I don't have a fix for `ReadHolds` being a quality-of-life struct that doesn't know what to do if its invariants aren't met.
